tools/doxygen.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Copyright 2023 Ekumen, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generate Beluga API documentation using doxygen.
+
+[[ -z "${WITHIN_DEV}" ]] && echo -e "\033[1;33mWARNING: Try running this script inside the development container if you experience any issues.\033[0m"
+
+set -o errexit -o xtrace
+
+cd $(dirname "$(readlink -f -- "$0")")/../beluga/docs
+doxygen ./Doxyfile
```

Comment:
@nahueespinosa meta: have you considered [adding a custom target](https://cmake.org/cmake/help/latest/command/add_custom_target.html) to CMake for this? It can sit behind a `BUILD_DOCS` option, or not.

Reply:
Done in https://github.com/Ekumen-OS/beluga/pull/197/commits/471ac152cc139961331d9c0cf4a49a35e807aec9.